Sources/TuistSupportTesting/Utils/MockSystem.swift

```diff
@@ -38,6 +38,7 @@ public final class MockSystem: Systeming {
         if stub.exitstatus != 0 {
             throw TuistSupport.SystemError.terminated(command: arguments.first!, code: 1)
         }
+        calls.append(command)
```

Comment:
What do we need the `call.append` cals for?

Reply:
I wanted to verify, that a system command was called. There's a utility for that:

```swift
MockSystem.called(_ args: String...) -> Bool
```
Unfortunately, the assertion was failing. I did a quick look, and the called commands in the mock where never added to the `calls` array. So I provided a fix 😊

Reply:
The way `MockSystem` is expected to be used is the following:
- Use these [helpers](https://github.com/tuist/tuist/blob/master/Sources/TuistSupportTesting/Utils/MockSystem.swift#L17) to define the execution of each of the expected commands.
-  Run the method under testing.

If the stubs are well defined, the test should run, otherwise it'll fail because `MockSystem` will fail the command.

Reply:
Hmm, so you think that it is not needed to double check if the expected command has been executed? 🤔 I cannot find any examples of assertions like that in the tests:

```
XCTAssertTrue(system.called(...))
```

that's why I suspect the `called(_: String...) -> Bool` method does not work as expected.

Reply:
> Hmm, so you think that it is not needed to double check if the expected command has been executed?

Exactly. when the test is executed, if the method is stubbed the test should pass, otherwise `MockSystem` will fail the command and the test will fail. There's no need to check whether the system was called.